camel/benchmarks/nexus.py

```diff
@@ -0,0 +1,481 @@
+                        result = compare_function_calls(
+                            agent_call=agent_call,
+                            ground_truth_call=ground_truth_call,
+                        )
+                        self._results.append(
+                            {
+                                "input": sample.input,
+                                "agent_call": agent_call,
+                                "ground_truth_call": ground_truth_call,
+                                "result": result,
+                                "error": None,
+                            }
+                        )
+                except Exception as e:
+                    logger.warning(f"Error in processing task: {sample.input}")
+                    self._results.append(
+                        {
+                            "input": sample.input,
+                            "agent_call": None,
+                            "ground_truth_call": ground_truth_call,
+                            "result": 0,
+                            "error": str(e),
+                        }
+                    )
+
+                agent.reset()
+
+                f.write(json.dumps(self._results[-1], indent=2) + "\n")
+                f.flush()
+
+        total = len(self._results)
+        correct = sum(r["result"] for r in self._results)
+
+        return {
+            "total": total,
+            "correct": correct,
+            "accuracy": correct / total,
+        }
+
+
+# Utility functions
+def construct_tool_descriptions(dataset_name) -> str:
+    r"""
+    Construct tool descriptions from function definitions and descriptions.
+    """
+    if dataset_name == "NVDLibrary":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="CVECPE"
+        )["train"]
+    elif dataset_name == "VirusTotal":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="VirusTotal"
+        )["train"]
+    elif dataset_name == "PlacesAPI":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="Places"
+        )["train"]
+    elif dataset_name == "ClimateAPI":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="Climate"
+        )["train"]
+    elif dataset_name == "OTX":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="OTX"
+        )["train"]
+    elif dataset_name == "VirusTotal-NestedCalls":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="VT_Multi (Nested)"
+        )["train"]
+    elif dataset_name == "VirusTotal-ParallelCalls":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="VT_Multi (Parallel)"
+        )["train"]
+    elif dataset_name == "NVDLibrary-NestedCalls":
+        dataset = load_dataset(
+            "Nexusflow/Function_Call_Definitions", name="CVECPE_Multi (Nested)"
```

Comment:
Hi, this block could be simplified by using a dictionary to map `dataset_name` to the corresponding `name` parameter. 

For example:

```python
HASH = {"NVDLibrary":"CVECPE", ...}

dataset = load_dataset(
            "Nexusflow/Function_Call_Definitions", name=HASH[dataset_name]
        )["train"]

```